File: getshells.py
#!/usr/bin/env python3

from collections import defaultdict
shellcnt = defaultdict(int)

with open('passwd', "r") as pw:
    for line in pw.readlines():
        shell = line[line.rfind(':')+1:].strip()
        shellcnt[shell] += 1

shells = sorted(shellcnt.keys())

for i in shells:
    print("{0:18} {1:5}".format(i, shellcnt[i]))

File: getshells.jl
#!/usr/bin/env julia

shellcnt = Dict{String,Int64}()

try
  open("passwd", "r") do f
  while ! eof(f)
    s = readline(f)
    pwline = split(s, ":")
    shell = pwline[7]
    shellcnt["$shell"] = get!(shellcnt, "$shell", 0) +1
  end
end
catch
    println("file not found")
end

for i in keys(shellcnt)
  println(i, ":\t", shellcnt[i])
end


